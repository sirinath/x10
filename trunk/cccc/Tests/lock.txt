# Test lock-release-acquire scenario.

0 0 w x 1
0 1 w l 0
1 0 r l 0 # might need atomic op...
1 1 w l 1
1 2 r x 0

0.1 -> 1.0
1.2 -> 0.0
