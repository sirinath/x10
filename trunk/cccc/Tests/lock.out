0 0 w x 1
0 1 w l 0
1 0 r l 0
1 1 w l 1
1 2 r x 0
# Manually specified dependencies
0.1 -> 1.0
1.2 -> 0.0
# Program-order dependencies
0.0 -> 0.1
1.0 -> 1.1
1.0 -> 1.2
1.1 -> 1.2
# Propagated dependencies
0.0 -> 1.0
0.1 -> 1.1
0.1 -> 1.2
1.0 -> 0.0
1.1 -> 0.0
1.2 -> 0.1
Happens-before dependencies for task 0
0.0 -> 0.1
0.1 -> 1.1
1.1 -> 0.0
Cycle located:
0.0 -> 0.1  (program order)
0.1 -> 1.1  (propagation)
1.1 -> 0.0  (propagation)
Task 0 sees cycle
Happens-before dependencies for task 1
0.0 -> 0.1
0.0 -> 1.0
0.1 -> 1.0
0.1 -> 1.1
0.1 -> 1.2
1.0 -> 0.0
1.0 -> 1.1
1.0 -> 1.2
1.1 -> 0.0
1.1 -> 1.2
1.2 -> 0.0
1.2 -> 0.1
Cycle located:
0.0 -> 0.1  (program order)
0.1 -> 1.0  (manual)
1.0 -> 0.0  (propagation)
Task 1 sees cycle
Cycle found: illegal CCCC execution
